### DISCUSSION TITLE
Refresh State

### DISCUSSION THREAD

**Discussion by ikaheraJQ** [Budibase/budibase#5626] [Help] [answered]
Hello everyone. I am sorry, I want to ask you guys about refreshing state.

I need to change State X that needed to show data on page A. State X is changed from a modal. The detail is everytime I click Apply Button on the modal, State X will change into defined value, the modal is closed, and then page A will automatically reload the data based on the updated value of State X. But the current situation is I need to refresh manually, for page A, to update the data everytime I change State X from the modal. Is it possible to automatically refresh State X value without manually refresh on page A? 

Thank you for any kind of help 🙏🏽

**Comment by melohagan** [comment, marked as answer]
Hi @ikaheraJQ 

I recently had this problem. See the bottom of this answer: https://github.com/Budibase/budibase/discussions/5524#discussioncomment-2614760

I think a new feature is needed for this. I.e. the redirect I had suggested, or some other way to update state dynamically from a modal

**Reply by ikaheraJQ** [reply]
Okay @melohagan, thanks for your help 🙏🏽

**Reply by melohagan** [reply]
Hey @ikaheraJQ I've just raised an issue here: https://github.com/Budibase/budibase/issues/5629

May be a while before it's worked on, or there may be other alternatives put forward.
